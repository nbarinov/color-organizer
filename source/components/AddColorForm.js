import React from 'react';

import '../style/add-color-form.css';

const AddColorForm = ({ className, onNewColor = f => f }) => {
    let _title, _color;

    const submit = e => {
        e.preventDefault();

        onNewColor(_title.value, _color.value);

        _title.value = '';
        _color.value = '#000000';
        _title.focus();
    };

    return (
        <form onSubmit={submit} className={(className) ? className + ' add-color-form' : 'add-color-form'}>
            <input type="text" className="add-color-form__input" ref={input => _title = input} placeholder="Color title..." required />
            <input type="color" className="add-color-form__color" ref={input => _color = input} required />
            <button className="add-color-form__button">Add new color</button>
        </form>
    );
};

export default AddColorForm