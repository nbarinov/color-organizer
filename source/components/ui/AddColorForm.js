import { Component } from 'react';
import PropTypes from 'prop-types';

import '../../style/add-color-form.css';

const AddColorForm = ({ className, nNewColor = f => f }) => {
    let _title, _color;

    const submit = e => {
        e.preventDefault();

        onNewColor(_title.value, _color.value);

        _title.value = '';
        _color.value = '#000000';
        _title.focus();
    };

    return (
        <form onSubmit={submit} className={(className) ? className + ' add-color-form' : 'add-color-form'}>
            <h2 className='add-color-form__title'>Add Color</h2>
            <div className="add-color-form__fields">
                <input type="text" className="add-color-form__input" ref={input => _title = input} placeholder="Color title..." required />
                <input type="color" className="add-color-form__color" ref={input => _color = input} required />
                <button className="add-color-form__button">Add new color</button>
            </div>
        </form>
    );
};

AddColorForm.propTypes = {
    onNewColor: PropTypes.func,
};

export default AddColorForm