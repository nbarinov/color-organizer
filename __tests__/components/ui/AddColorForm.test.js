import AddColorForm from '../../../source/components/ui/AddColorForm';

const { mount, shallow } = Enzyme;

describe("<AddColorFrom /> UI Component", () => {
    let wrapper;

    describe("Rendering for UI", () => {
        it("renders correct properties", () => {
            expect(
                shallow(<AddColorForm />).html()
            ).toMatchSnapshot();
        });

        it("renders parental className add-color-form", () =>
            expect(
                shallow(<AddColorForm className="parent__form" />)
                    .find('form.parent__form.add-color-form')
                    .length
            ).toBe(1)
        );

        it("Submitting default onNewColor does not cause error", () => {
            wrapper = mount(<AddColorForm />);
            wrapper.ref('_title').value = 'test-color';
            wrapper.ref('_color').value = '#0000FF';
            wrapper.find('form').simulate('submit');
        });
    });

    describe("Adding new Colors", () => {
        const _addColor = jest.fn();

        beforeAll(() => {
            wrapper = mount(<AddColorForm onNewColor={_addColor} />);
            wrapper.ref('_title').value = 'test-color';
            wrapper.ref('_color').value = '#000099';
            wrapper.find('form').simulate('submit');
        });

        it("adds colors correctly", () =>
            expect(_addColor).toBeCalledWith('test-color', '#000099')
        );

        it("resets the title value", () =>
            expect(wrapper.ref('_title').value).toBe('')
        );

        it("resets the color value", () =>
            expect(wrapper.ref('_color').value).toBe("#000000")
        );
    });
});