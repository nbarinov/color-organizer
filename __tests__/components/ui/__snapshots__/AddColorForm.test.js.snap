// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddColorFrom /> UI Component Rendering for UI renders correct properties 1`] = `"<form class=\\"add-color-form\\"><h2 class=\\"add-color-form__title\\">Add Color</h2><div class=\\"add-color-form__fields\\"><input type=\\"text\\" class=\\"add-color-form__input\\" placeholder=\\"Color title...\\" required=\\"\\"/><input type=\\"color\\" class=\\"add-color-form__color\\" required=\\"\\"/><button class=\\"add-color-form__button\\">Add new color</button></div></form>"`;
